### PR TITLE
plugin/ping: Transition from defined type to class

### PR DESCRIPTION
**Pull request by daenney**
This brings the ping plugin inline with the rest of the plugins. It transitions it from a defined type that could be called multiple times creating multiple `<Plugin ping></Plugin>`, definitions which
appears not to be valid, to a class.

As one can pass in an array to hosts you can still ping multiple targets.

### CODE REVIEW

**Comment by daenney** [comment]
@nibalizer Can you fix PCCI please? The reported URL for the failed build log doesn't work.

**Comment by daenney** [comment]
Also, Beaker definition is broken for PCCI and CentOS 7:

```
/home/pcci/new_ruby_gems_home/gems/beaker-2.14.1/lib/beaker/dsl/install_utils/foss_utils.rb:250:in `block in install_puppet_on': install_puppet() called for unsupported platform 'centos-7-1505-x86_64' on 'centos-7-1505-x86_64' (RuntimeError)
	from /home/pcci/new_ruby_gems_home/gems/beaker-2.14.1/lib/beaker/shared/host_manager.rb:104:in `run_block_on'
	from /home/pcci/new_ruby_gems_home/gems/beaker-2.14.1/lib/beaker/shared/host_manager.rb:93:in `block in run_block_on'
	from /home/pcci/new_ruby_gems_home/gems/beaker-2.14.1/lib/beaker/shared/host_manager.rb:92:in `map'
	from /home/pcci/new_ruby_gems_home/gems/beaker-2.14.1/lib/beaker/shared/host_manager.rb:92:in `run_block_on'
	from /home/pcci/new_ruby_gems_home/gems/beaker-2.14.1/lib/beaker/dsl/patterns.rb:32:in `block_on'
	from /home/pcci/new_ruby_gems_home/gems/beaker-2.14.1/lib/beaker/dsl/install_utils/foss_utils.rb:231:in `install_puppet_on'
	from /home/pcci/new_ruby_gems_home/gems/beaker-2.14.1/lib/beaker/dsl/install_utils/foss_utils.rb:178:in `install_puppet'
	from /tmp/tmpBr_vm_/job/spec/spec_helper_acceptance.rb:6:in `block in <top (required)>'
	from /tmp/tmpBr_vm_/job/spec/spec_helper_acceptance.rb:4:in `each'
	from /tmp/tmpBr_vm_/job/spec/spec_helper_acceptance.rb:4:in `<top (required)>'
	from /usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require'
	from /usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require'
	from /tmp/tmpBr_vm_/job/spec/acceptance/class_spec.rb:1:in `<top (required)>'
	from /home/pcci/new_ruby_gems_home/gems/rspec-core-3.3.1/lib/rspec/core/configuration.rb:1327:in `load'
	from /home/pcci/new_ruby_gems_home/gems/rspec-core-3.3.1/lib/rspec/core/configuration.rb:1327:in `block in load_spec_files'
	from /home/pcci/new_ruby_gems_home/gems/rspec-core-3.3.1/lib/rspec/core/configuration.rb:1325:in `each'
	from /home/pcci/new_ruby_gems_home/gems/rspec-core-3.3.1/lib/rspec/core/configuration.rb:1325:in `load_spec_files'
	from /home/pcci/new_ruby_gems_home/gems/rspec-core-3.3.1/lib/rspec/core/runner.rb:102:in `setup'
	from /home/pcci/new_ruby_gems_home/gems/rspec-core-3.3.1/lib/rspec/core/runner.rb:88:in `run'
	from /home/pcci/new_ruby_gems_home/gems/rspec-core-3.3.1/lib/rspec/core/runner.rb:73:in `run'
	from /home/pcci/new_ruby_gems_home/gems/rspec-core-3.3.1/lib/rspec/core/runner.rb:41:in `invoke'
	from /home/pcci/new_ruby_gems_home/gems/rspec-core-3.3.1/exe/rspec:4:in `<top (required)>'
	from /home/pcci/new_ruby_gems_home/bin/rspec:23:in `load'
	from /home/pcci/new_ruby_gems_home/bin/rspec:23:in `<main>'
```

**Comment by igalic** [comment]
@daenney yes, beaker is still broken, there's a way to fix it, but @nibalizer didn't have time to modulesync that yet

File: templates/plugin/ping.conf.erb [hunk 1]
```diff
@@ -1,27 +1,23 @@
-# Generated by Puppet
-LoadPlugin "ping"
```
Comment (blkperl) [inline templates/plugin/ping.conf.erb]: Does this still work without this load line?

Comment (daenney) [inline templates/plugin/ping.conf.erb]: Yes. It's automatically added on to it by the `collected::plugin` defined type. It includes that part of the template: https://github.com/puppet-community/puppet-collectd/blob/master/manifests/plugin.pp

**Comment by daenney** [comment]
Ding dong @igalic @blkperl @nibalizer

**Comment by igalic** [comment]
thank @daenney sorry for so late i busy

**Comment by daenney** [comment]
That's OK :heart:

**Pull request merged**